src/pugixml.cpp

```diff
@@ -9950,7 +9966,7 @@ PUGI_IMPL_NS_BEGIN
 
 			xpath_node* last = ns.begin() + first;
 
-			xpath_context c(xpath_node(), 1, size);
+			xpath_context c = xpath_context(xpath_node(), 1, size);
```

Comment:
Does it compile like this? This would be a little closer to the rest of the code, and avoid a redundant copy.

```
xpath_node cn;
xpath_context c(cn, 1, size);
```

Reply:
Yes, it works.